File: exercism/go/isogram/isogram.go
package isogram

func IsIsogram(word string) bool {
	letterMask := 0
	for i := range word {
		letter := word[i]
		if letter == ' ' || letter == '-' {
			continue
		}
		if letter >= 'A' && letter <= 'Z' {
			letter += 'a' - 'A' // lowercase
		}
		letter -= 'a' // normalize
		if (letterMask & (1 << letter)) != 0 {
			return false
		}
		letterMask |= (1 << letter)
	}
	return true
}

File: exercism/c/isogram/isogram.c
#include "isogram.h"

#include <stdio.h>

bool is_isogram(const char phrase[]) {
  if (!phrase)
    return false;

  unsigned int letters = 0; // a bitmap
  for (const char *c = phrase; *c; c++) {
    if (*c == ' ' || *c == '-')
      continue;
    const int letterMask = 1 << (*c - 'a');
    if (letters & letterMask)
      return false;
    letters |= letterMask;
  }
  return true;
}
